app/controllers/users_controller.rb

```diff
@@ -0,0 +1,17 @@
+class UsersController < ApplicationController
+  def index
+    fetch_current_user_details
+    render json: {
+      current_user: current_user,
+      current_user_roles: @current_user_roles,
+      users: User.all
+    }
+  end
+
+private
+
+  def fetch_current_user_details
+    user = current_user
```

Comment:
Why do you create new instance of user model here, but uses default in line number 5?